docs/dg/dev/best-practices/project-nfr-templates/monitorable-process-guidelines.md

```diff
@@ -0,0 +1,253 @@
+* Silenced exceptions: Instances of an exception being consumed by an empty `catch()` clause.
+* Significant user actions and events:
+    * Help requests
+    * Cancelled actions
+* Notable user actions:
+    * Successful logins
+    * Privilege elevations
+    * Data validation failures
+    * Payment and transaction events
+    * Significant user journey events
+
+### What not to log?
+* Unreasonable logging: Avoid logging without a clear purpose. When deciding to create a log entry, identify at least one use case for the data.
+* Performance critical places: Minimize logging in areas where performance is critical. If there's a performance-critical process or a large cycle as a sub-process element, try to strike a balance between performance and traceability.
+* Audit trail: While it's recommended to log the audit trails of significant entity changes, these logs shouldn't be sent to the regular log system due to the sensitive nature of the data typically involved in authentication and authorization.
+
+### Log levels
+Appropriately implement the following log levels in application workflows to simplify issue investigations and resolutions. The log levels are based on Syslog [RFC 5424](https://datatracker.ietf.org/doc/html/rfc5424).
+
+| Level | Description |
+|-------|-------------|
+| Debug | Provides detailed data that's mostly used for debugging. These logs allow to check variable values and/or error stacks. |
+| Info |  Provides information about the normal operation of the application. This level is often used to record the progress of a task or operation. |
+| Notice | Provides information about normal, but significant, events that occur within the application. |
+| Warning | Describes events that are less destructive than errors. They usually don't result in any reduction of the system's functionality or its full failure. Undesirable things that are not necessarily wrong. |
+| Error | Identifies error events that may still let the software run but with restricted capabilities in the impacted routes. |
+| Critical | Identifies extremely serious error events that are likely to cause the system to abort. Typically, this leads to catastrophic failures. |
+| Alert | Indicates a situation that requires immediate attention. |
+| Emergency | Indicates a severe problem that requires immediate action to prevent damage or loss. |
+
+### Log entry format
+A log entry must always answer the following items and follow the describes structure:
+* *who* caused an event.
+* *when* exactly did the event happen.
+* *where* did the event take place, for example-context, component or application.
+* *what* happened and/or why it happened.
+* *result* of event: exception, success details, information details.
+
+Log structure example (consider using [CloudWatch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-metric-streams-formats-json.html) and/or [Cloud Handler](https://github.com/maxbanton/cwh) recommendations):
```

Comment:
How's the content in braces related to the log structure example?

Reply:
In general, the reader of this document needs to settle on a log structure. Here below, i'm providing a log structure as an example/base for designing their project specific log structure. To best design the project specific log structure, they should consider the CloudWatch's log structure, and/or additionnaly the Cloud Handler and decide what to take from each of these.